gossipsub.go

```diff
@@ -208,21 +230,29 @@ func (gs *GossipSubRouter) handlePrune(p peer.ID, ctl *pb.ControlMessage) {
 	}
 }
 
-func (gs *GossipSubRouter) Publish(from peer.ID, msg *pb.Message) {
-	gs.mcache.Put(msg)
-
-	tosend := make(map[peer.ID]struct{})
+func (gs *GossipConfigurableRouter) Publish(from peer.ID, msg *pb.Message) {
 	for _, topic := range msg.GetTopicIDs() {
+		config, ok := gs.topicConfigs[topic]
```

Comment:
that's just backwards; can't have a `config` object handling hte most important function of the router!

Reply:
Isn't that the whole premise of every inversion of control system? We could create new routers and pubsub instances for every configuration option instead of using one pubsub instance.

However, that means
1. having multiple heartbeat tickers running at the same time
2. not reusing any of the gossipsub structures across implementations 
3. ending up with weird multiple tenancy issues unless we change PubSub more. For example, if right now you create a FloodSub and GossipSub instance on the same node you end up with overly excessive message passing *because* GossipSub is backwards compatible with FloodSub and will handle both messages. Either we need one router to handle multiple configurations or we need to make PubSub able to handle backwards compatible protocols, this way involves less protocol work.